### PR TITLE
Closes #8769: Migrate StudioPress accelerator into Hostings Subscriber

### LINKED ISSUE #8769: Migrate legacy theme compatibility into the ThirdParty service provider
> 🤖 AI-generated — created by an automated pipeline. Review before acting on this.

**Context**
This is preparation work for #6418 ("Implement a plugin resolver service to optimize loading of 3rd party compatibility subscribers"). 

The optimization in #6418 will add a resolver that detects the environment early and only instantiates the 3rd-party compatibility subscribers that are actually needed, mirroring the existing `ThemeResolver` + `SubscriberFactory` pattern under `inc/ThirdParty/Themes/`. However, the legacy **theme** compatibility still lives as a procedural file under `inc/3rd-party/themes/`, `require`d unconditionally by `inc/3rd-party/3rd-party.php`, so the resolver has no coverage over it.

This ticket is the prerequisite refactoring to bring legacy theme compatibility into the structured ThirdParty container, enabling the resolver to optimize loading.

**Scope**
- `inc/3rd-party/themes/studiopress.php` — currently required unconditionally. It hooks `admin_init` (`rocket_clear_cache_after_studiopress_accelerator`) and `rocket_after_clean_domain` (`rocket_clean_studiopress_accelerator`) to integrate with the StudioPress / Genesis Accelerator cache.

**Migration Approach**
1. Refactor the procedural file (top-level `add_action` + functions) into a Subscriber class exposing `get_subscribed_events()`, following existing `inc/ThirdParty/Themes/*` subscribers.
2. Register it through the Themes service provider and, if a reliable detection signal exists for StudioPress/Genesis Accelerator, wire it via `ThemeResolver` / `SubscriberFactory` so it is only instantiated when relevant. **Open question:** this file currently has no detection condition — decide whether a proper gate can be added or whether it must stay always-loaded.
3. Preserve current behavior exactly — same hooks, same cache-clearing behavior.
4. Remove the migrated `require` line from `inc/3rd-party/3rd-party.php` once ported.

**Acceptance Criteria**
- [ ] StudioPress compatibility is a container-registered subscriber under the ThirdParty namespace and no longer unconditionally `require`d
- [ ] If a detection signal is viable, instantiation only occurs when relevant; otherwise the decision to keep it always-loaded is documented in code comments
- [ ] No behavioral change: cache clearing on StudioPress/Genesis Accelerator still works as before
- [ ] Existing tests pass; add coverage where practical

**Development Steps**
- [ ] Create `inc/ThirdParty/Themes/StudioPress.php` Subscriber class with `get_subscribed_events()`
- [ ] Move hook callbacks and integration logic from `inc/3rd-party/themes/studiopress.php` into the new subscriber
- [ ] Register subscriber in Themes service provider
- [ ] Evaluate and implement StudioPress/Genesis detection if viable; document decision if not
- [ ] Remove `require` directive from `inc/3rd-party/3rd-party.php`
- [ ] Run test suite to verify no regressions
- [ ] Add integration test coverage if not already present

**Effort Estimation**
S (< 1 day)

**Additional Information**
Grooming confidence: High
Related: #6418

### PR DESCRIPTION
**Pull request by remyperona**
> 🤖 AI-generated — created by an automated pipeline. Review before acting on this.

Closes #8769

# Description

Migrates the legacy `inc/3rd-party/themes/studiopress.php` procedural file into a container-registered Subscriber class at `inc/ThirdParty/Hostings/StudioPress.php`, resolving the issue where StudioPress/Genesis Accelerator compatibility was unconditionally `require`d outside the DI container. The integration now lives under the Hostings namespace and is registered via the container, while preserving identical runtime detection behavior and cache-clearing functionality.

## Type of change

- [ ] New feature (non-breaking change which adds functionality).
- [ ] Bug fix (non-breaking change which fixes an issue).
- [x] Enhancement (non-breaking change which improves an existing functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as before).
- [ ] Sub-task of #8769
- [ ] Chore
- [ ] Release

### What was tested

Automated coverage authored and run for the migrated StudioPress subscriber:

- **Unit** (`tests/Unit/inc/ThirdParty/Hostings/StudioPress/`): `getSubscribedEvents` (asserts both hooks are returned unconditionally), `clearCacheAfterAccelerator` (capability guard, nonce handling, URL-purge vs theme-purge branches), `cleanAcceleratorCache` (runtime `isset`/`is_a` guard, `cache_flush_theme()` invoked only when the global is present).
- **Integration** (`tests/Integration/inc/ThirdParty/Hostings/StudioPress/`): `hooksAreRegistered` (container resolves `studiopress_accelerator` to a `StudioPress` instance and both `admin_init` / `rocket_after_clean_domain` callbacks are actually attached after bootstrap), `cleanAcceleratorCache`.
- Both hook callbacks were verified byte-for-byte equivalent to the deleted procedural `inc/3rd-party/themes/studiopress.php`; no behavioral change.

**QA: PASS** (report: https://github.com/wp-media/wp-rocket/pull/8771#issuecomment-5430472762). QA booted the wp-env environment and verified all four acceptance criteria live: the container resolves `studiopress_accelerator` to a `StudioPress` instance; both `admin_init` and `rocket_after_clean_domain` hooks are attached at priority 10; a full cache clear ran with no warnings/fatals; and the new suites passed in-container (unit `--group StudioPress` 10 tests/41 assertions, integration `--group StudioPress` 3 tests/6 assertions). The real host-purge branch (`SP_Accel_Nginx_Proxy_Cache_Purge::cache_flush_theme()`) is guarded and not exercisable off a StudioPress host — covered by Mockery-based tests. All CI checks green (PHPUnit matrix PHP 7.4–8.5, PHPCS, PHPStan, task-check).

### How to test

1. **Manual Smoke Test (required):** On a site WITHOUT `SP_Accel_Nginx_Proxy_Cache_Purge` defined (the common case):
   - Verify the StudioPress subscriber hooks are still registered: `wp eval 'var_dump( has_action( "admin_init" ), has_action( "rocket_after_clean_domain" ) );'` should confirm both hooks attached.
   - Trigger a full cache clear via the admin dashboard "Clear cache" button, or programmatically with `wp eval 'rocket_clean_domain();'`.
   - Confirm cache clear completes without fatal errors or warnings referencing `sp_accel_nginx_proxy_cache_purge`.

2. **Automated Unit Test Coverage:** 10 new unit tests in `tests/Unit/inc/ThirdParty/Hostings/StudioPress/` cover:
   - `clear_cache_after_accelerator()` with: (a) capability denied, (b) valid `sp-accel-purge-url` nonce + `cache-purge-url` present, (c) valid `sp-accel-purge-theme` nonce, (d) invalid/missing nonce, (e) `$GLOBALS['sp_accel_nginx_proxy_cache_purge']` unset.
   - `clean_accelerator_cache()` with the global set and absent.

3. **Automated Integration Test Coverage:** 3 new integration tests in `tests/Integration/inc/ThirdParty/Hostings/StudioPress/` verify:
   - End-to-end hook wiring via the real `Subscriber_Interface` and event manager.
   - Subscriber is registered and `has_action()` reports both hooks present **regardless** of `HostResolver::get_host_service()`'s return value (locks in the "always-on" contract).

### Affected Features & Quality Assurance Scope

- StudioPress/Genesis Accelerator nginx cache-purge integration.
- WP Rocket cache-clearing workflows: admin dashboard "Clear cache" button, WP-CLI `wp rocket clean`, and programmatic `rocket_clean_domain()` calls.
- ThirdParty DI container registration and event subscriber wiring.
- Existing Kinsta/WPEngine/GoDaddy Hostings subscribers remain unaffected.

### Documentation

**Why always-loaded registration (not gated by `HostResolver`):**

`SP_Accel_Nginx_Proxy_Cache_Purge` can be defined by a Genesis/StudioPress child theme's `functions.php`, which WordPress loads **after** the `plugins_loaded` action. `HostResolver::get_host_service()` runs inside `Hostings\ServiceProvider::register()`, which is hooked to `plugins_loaded`. A resolver-time `class_exists()` check would return `false` on sites where the integration exists but is defined theme-side, silently dropping both hooks on exactly the sites this integration targets — a regression from the current always-loaded `require`.

Detection is therefore performed at hook-fire time inside each callback (`admin_init` and `rocket_after_clean_domain`), using the runtime `isset()`/`is_a()` guard that was already present in the legacy code. This guard is evaluated after the theme has loaded, so it correctly detects theme-defined instances.

**Implementation pattern:**

- `StudioPress` class implements `Subscriber_Interface` and is registered as a shared container binding (`'studiopress_accelerator' => StudioPress::class`) unconditionally in `Hostings\ServiceProvider::register()`.
- `get_subscribed_events()` returns both hooks unconditionally (no resolver-time gate).
- Each callback method preserves the original `isset()`/`is_a()` guard verbatim; mirrors the pattern used by `Kinsta.php` (defensive re-check even though the subscriber is conditionally registered there; here the guard is the *entire* detection mechanism).
- `StudioPress` is added to the `$common_subscribers` array in `Plugin::init_common_subscribers()` as an unconditional entry, matching the pattern used by `'jetpack'`, `'mobile_subscriber'`, and `'wordfence_subscriber'`.
- `HostResolver.php` and `HostSubscriberFactory.php` are left unchanged; no new detection branch or case added.

**Behavioral equivalence:**

Both hooks (`admin_init` for user-initiated purge confirmation, `rocket_after_clean_domain` for automatic cache coordination) are ported verbatim from the legacy procedural file:
- `clear_cache_after_accelerator()` → previously `rocket_clear_cache_after_studiopress_accelerator()`
- `clean_accelerator_cache()` → previously `rocket_clean_studiopress_accelerator()`

All conditionals, nonce action strings, sanitization, and function calls are unchanged.

### New dependencies

None.

### Risks

**None identified.**

The new code path is a 1:1 refactoring of existing, production-proven logic. The only structural change is moving from a procedural `require` to a container-registered Subscriber, which is the standard DI pattern used throughout the codebase. Runtime behavior is identical (detection and cache-clearing happen at the same hooks with the same guards).

No new permissions, capabilities, or API calls introduced. Multisite network activation and plugin activation/deactivation handling are unaffected (the legacy code never hooked those events either).

### Follow-up tickets

None. Two nice-to-have items surfaced during review (a wiring-key regression unit test, and a doc comment on the always-on integration test's coverage limitation) were reviewed and discarded — the existing integration test already asserts the end-to-end wiring at runtime.

# Mandatory Checklist

## Code validation

- [x] I validated all the Acceptance Criteria. If possible, provide screenshots or videos.
- [x] I triggered all changed lines of code at least once without new errors/warnings/notices.
- [x] I implemented built-in tests to cover the new/changed code.

## Code style

- [x] I wrote a self-explanatory code about what it does.
- [x] I protected entry points against unexpected inputs.
- [x] I did not introduce unnecessary complexity.
- [ ] Output messages (errors, notices, logs) are explicit enough for users to understand the issue and are actionnable.

## Unticked items justification

Output-messages item is N/A: this refactoring introduces no user-facing output messages — it relocates existing hook callbacks 1:1 with no new notices, errors, or logs. Acceptance criteria were validated by the QA pass (see "What was tested"); the new/changed lines are exercised by the unit + integration suites.

# Additional Checks

- [x] In the case of complex code, I wrote comments to explain it.
- [ ] When possible, I prepared ways to observe the implemented system (logs, data, etc.).
- [ ] I added error handling logic when using functions that could throw errors (HTTP/API request, filesystem, etc.)

### CODE REVIEW

**Comment by remyperona** [comment]
> [!NOTE]
> Generated by the AI delivery pipeline (qa-engineer · Claude Opus 4.8).

**QA: ✅ PASS**

| Acceptance Criterion | Method | Result |
|---|---|---|
| Container-registered subscriber under ThirdParty namespace, legacy `require` removed | API/Analysis | ✅ |
| Detection at hook-fire time; always-loaded decision documented in code | Analysis | ✅ |
| No behavioral change: cache clearing on StudioPress/Genesis Accelerator still works | API/Analysis | ✅ (host-specific purge path guarded — see note) |
| Existing tests pass; new unit + integration coverage added | API (local phpunit) | ✅ |

**Evidence**

1. **Container registration** — `docker exec ... wp eval` confirms `apply_filters('rocket_container','')->has('studiopress_accelerator')` returns `true`, resolving to `WP_Rocket\ThirdParty\Hostings\StudioPress`. `inc/ThirdParty/Hostings/ServiceProvider.php:56` calls `addShared('studiopress_accelerator', StudioPress::class)` unconditionally, ahead of and independent from the `HostResolver::get_host_service()` gate that applies only to the actual hosting subscriber (lines 58-63). `inc/3rd-party/3rd-party.php` no longer requires `themes/studiopress.php` (deleted, confirmed via `git diff 308c4d793`).

2. **No resolver-time gate** — `inc/ThirdParty/Hostings/StudioPress.php:5-16` docblock documents the rationale (child-theme `functions.php` loads after `plugins_loaded`, so a resolver-time `class_exists()` would miss theme-defined instances). Grep for `class_exists`/`get_host_service` in `StudioPress.php` returns no hits — confirms no resolver-time gate. `inc/Plugin.php:426` adds `'studiopress_accelerator'` to the unconditional `$common_subscribers` array (same tier as `jetpack`, `optimole_subscriber`), above the `HostResolver::get_host_service()` conditional block.

3. **Hook wiring + cache clear** — `wp eval` confirms both `has_action('admin_init', [$sub,'clear_cache_after_accelerator'])` and `has_action('rocket_after_clean_domain', [$sub,'clean_accelerator_cache'])` return `10` (registered). Running `rocket_clean_domain()` via `wp eval --debug` completes with no PHP warnings/notices/fatals and no `debug.log` entries. Note: the PR's suggested `wp rocket clean --confirm` command does not exist in this codebase (`wp help rocket` only lists `abilities-catalog`); I used `rocket_clean_domain()` directly, which fires the same `rocket_after_clean_domain` hook.
   **Guard note (CANNOT_VERIFY in this environment, as expected):** the actual StudioPress-host purge call (`$GLOBALS['sp_accel_nginx_proxy_cache_purge']->cache_flush_theme()`) is gated by `isset($GLOBALS['sp_accel_nginx_proxy_cache_purge']) && is_a(..., 'SP_Accel_Nginx_Proxy_Cache_Purge')` at `inc/ThirdParty/Hostings/StudioPress.php:46,72`. Neither the global nor the `SP_Accel_Nginx_Proxy_Cache_Purge` class exist on this local/non-StudioPress environment, so the purge branch is a verified no-op here, not a positive confirmation of the real hosting integration. This path is covered by the unit/integration tests via Mockery mocks of the class instead.

4. **Test coverage** — ran the new suites directly inside the wp-env `cli` container (not CI, but a real local execution):
   - `vendor/bin/phpunit --configuration tests/Unit/phpunit.xml.dist --group StudioPress` → **10 tests, 37 assertions, OK**.
   - `vendor/bin/phpunit --configuration tests/Integration/phpunit.xml.dist --group StudioPress` → **3 tests, 6 assertions, OK**.
   Read all new test/fixture files — coverage matches what the PR describes (capability denied, valid `sp-accel-purge-url` nonce with URL, valid `sp-accel-purge-theme` nonce, invalid nonce, missing global, valid-nonce-but-empty-url edge case, and an integration test locking in the "always registered regardless of `HostResolver`" contract).

**Smoke tests**
- Plugin deactivate/reactivate cycle: clean, no fatals/warnings.
- `/wp-admin/options-general.php?page=wprocket` and `/wp-admin/` load with HTTP 200 (authenticated) and no `debug.log` output.

**Note on scope vs. originating issue:** issue #8769 suggested `inc/ThirdParty/Themes/StudioPress.php` registered via the Themes service provider / `ThemeResolver`. The PR instead places the class under `inc/ThirdParty/Hostings/` and registers it in the Hostings `ServiceProvider`, with a documented rationale (mirrors the `Kinsta.php` pattern; StudioPress/Genesis Accelerator is a hosting-infra integration, not a theme-detection one). This satisfies the literal acceptance criteria ("container-registered subscriber under the ThirdParty namespace") and is a reasonable interpretation, but it is a location deviation from the issue text worth a reviewer's explicit sign-off.

File: tests/Unit/inc/ThirdParty/Hostings/StudioPress/cleanAcceleratorCache.php [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+		parent::tearDown();
+	}
+
+	/**
+	 * @dataProvider configTestData
+	 */
+	public function testShouldReturnAsExpected( $config ) {
+		if ( $config['global_set'] ) {
+			$accelerator = Mockery::mock( 'SP_Accel_Nginx_Proxy_Cache_Purge' );
+			$accelerator->expects()->cache_flush_theme();
```
Comment (remyperona) [inline tests/Unit/inc/ThirdParty/Hostings/StudioPress/cleanAcceleratorCache.php:40]: [MEDIUM] PHPStan CI is red because of this line: `Call to method expects() on an unknown class SP_Accel_Nginx_Proxy_Cache_Purge` (`method.internal`/`class.notFound`, confirmed in the `lint / PHPStan` check run). `SP_Accel_Nginx_Proxy_Cache_Purge` is never declared anywhere in the codebase (it is a purely runtime, theme/hosting-provided class), so PHPStan cannot resolve the Mockery mock target.

**Fix:** Add a fixture stub class, mirroring the existing pattern at `tests/Fixtures/inc/ThirdParty/Hostings/WPEngine/WpeCommon.php` (guarded by `class_exists()`), e.g. `tests/Fixtures/inc/ThirdParty/Hostings/StudioPress/SPAccelNginxProxyCachePurge.php` declaring a minimal `SP_Accel_Nginx_Proxy_Cache_Purge` class with a `cache_flush_theme()` method. `tests/Fixtures/` is already in `phpstan.neon.dist`'s `scanDirectories`, so this resolves the error for both the unit and integration test.

File: tests/Integration/inc/ThirdParty/Hostings/StudioPress/cleanAcceleratorCache.php [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+	public function tear_down() {
+		unset( $GLOBALS['sp_accel_nginx_proxy_cache_purge'] );
+
+		parent::tear_down();
+	}
+
+	public function testShouldCallCacheFlushThemeWhenAcceleratorPresent() {
+		$accelerator = Mockery::mock( 'SP_Accel_Nginx_Proxy_Cache_Purge' );
+		$accelerator->expects()->cache_flush_theme();
```
Comment (remyperona) [inline tests/Integration/inc/ThirdParty/Hostings/StudioPress/cleanAcceleratorCache.php:24]: [MEDIUM] Same PHPStan CI failure as the unit test counterpart: `Call to method expects() on an unknown class SP_Accel_Nginx_Proxy_Cache_Purge`.

**Fix:** Same fix as flagged on `tests/Unit/.../cleanAcceleratorCache.php` — add a `tests/Fixtures/inc/ThirdParty/Hostings/StudioPress/*.php` stub class for `SP_Accel_Nginx_Proxy_Cache_Purge` (mirroring `WpeCommon.php`'s pattern), which will resolve this occurrence too since it lives in the scanned `tests/Fixtures/` directory.

File: tests/Unit/inc/ThirdParty/Hostings/StudioPress/clearCacheAfterAccelerator.php [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+	 */
+	public function testShouldReturnAsExpected( $config ) {
+		Functions\expect( 'current_user_can' )
+			->with( 'rocket_manage_options' )
+			->once()
+			->andReturn( $config['has_cap'] );
+
+		if ( ! $config['has_cap'] ) {
+			$this->subscriber->clear_cache_after_accelerator();
+			$this->addToAssertionCount( 1 );
```
Comment (remyperona) [inline tests/Unit/inc/ThirdParty/Hostings/StudioPress/clearCacheAfterAccelerator.php]: [MEDIUM] PHPStan CI failure: `Call to internal method PHPUnit\Framework\TestCase::addToAssertionCount() from outside its root namespace PHPUnit` (`method.internal`). This exact pattern exists twice elsewhere in the suite but is only tolerated there because both are pinned by file+line in `phpstan-baseline.neon`; new occurrences are not covered and fail CI.

**Fix:** Replace `$this->addToAssertionCount( 1 );` with a real public assertion (e.g. `$this->assertTrue( true );` or, better, assert something concrete about the no-op path) so the internal-API call is avoided entirely — cleaner than adding new baseline entries. Same fix applies to the other two occurrences in this PR (`tests/Unit/.../cleanAcceleratorCache.php:48` and `tests/Integration/.../cleanAcceleratorCache.php:34`).

File: tests/Unit/inc/ThirdParty/Hostings/StudioPress/cleanAcceleratorCache.php [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+			$GLOBALS['sp_accel_nginx_proxy_cache_purge'] = $accelerator;
+		}
+
+		$this->subscriber->clean_accelerator_cache();
+
+		if ( ! $config['global_set'] ) {
+			$this->addToAssertionCount( 1 );
```
Comment (remyperona) [inline tests/Unit/inc/ThirdParty/Hostings/StudioPress/cleanAcceleratorCache.php]: [MEDIUM] Same PHPStan `method.internal` failure as flagged on `clearCacheAfterAccelerator.php:49` — `$this->addToAssertionCount( 1 )` calls an internal PHPUnit method from outside its namespace, uncovered by baseline for this new file.

**Fix:** Replace with `$this->assertTrue( true );` (or a concrete assertion).

File: tests/Integration/inc/ThirdParty/Hostings/StudioPress/cleanAcceleratorCache.php [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+		$GLOBALS['sp_accel_nginx_proxy_cache_purge'] = $accelerator;
+
+		do_action( 'rocket_after_clean_domain' );
+	}
+
+	public function testShouldDoNothingWhenAcceleratorAbsent() {
+		do_action( 'rocket_after_clean_domain' );
+
+		$this->addToAssertionCount( 1 );
```
Comment (remyperona) [inline tests/Integration/inc/ThirdParty/Hostings/StudioPress/cleanAcceleratorCache.php]: [MEDIUM] Same PHPStan `method.internal` failure as flagged on the unit tests — `$this->addToAssertionCount( 1 )` calls an internal PHPUnit method from outside its namespace, uncovered by baseline for this new file.

**Fix:** Replace with `$this->assertTrue( true );` (or a concrete assertion).

**Comment by remyperona** [comment]
> [!NOTE]
> Generated by the AI delivery pipeline (lead-reviewer · Claude Opus 4.8).

**Review: ✅ PASS**

Re-reviewed after commit `49f248169`. All 5 MEDIUM blockers from the prior pass are resolved:
- `tests/Fixtures/classes/SP_Accel_Nginx_Proxy_Cache_Purge.php` added (`class_exists()`-guarded stub, mirrors the existing `NinukisCaching.php` convention in the same directory) — resolves the `class.notFound` PHPStan errors.
- All 3 `addToAssertionCount(1)` calls replaced with real assertions (`assertArrayNotHasKey('sp_accel_nginx_proxy_cache_purge', $GLOBALS)` for the "global absent" no-op paths; explicit `Functions\expect(...)->never()` on all five downstream calls for the capability-denied early-return path) — resolves the `method.internal` PHPStan errors, and the capability-denied assertion is now strictly stronger than before.

Verified: `lint / PHPStan` and `lint / PHP CodeSniffer` CI checks now pass. Zero diff in `inc/` between `fd118a801` and `49f248169` — production logic (`StudioPress.php`, `ServiceProvider.php`, `Plugin.php`) is unchanged from the already-approved implementation. No new issues introduced.

File: tests/Unit/inc/ThirdParty/Hostings/StudioPress/cleanAcceleratorCache.php (hunk 1 above)
Comment (remyperona) [reply tests/Unit/inc/ThirdParty/Hostings/StudioPress/cleanAcceleratorCache.php:40]: Fixed in 49f248169 — added a PHPStan-resolvable stub `tests/Fixtures/classes/SP_Accel_Nginx_Proxy_Cache_Purge.php` and replaced the `addToAssertionCount()` calls with real assertions. Full-repo PHPStan now reports 0 errors.

File: tests/Integration/inc/ThirdParty/Hostings/StudioPress/cleanAcceleratorCache.php (hunk 2 above)
Comment (remyperona) [reply tests/Integration/inc/ThirdParty/Hostings/StudioPress/cleanAcceleratorCache.php:24]: Fixed in 49f248169 — added a PHPStan-resolvable stub `tests/Fixtures/classes/SP_Accel_Nginx_Proxy_Cache_Purge.php` and replaced the `addToAssertionCount()` calls with real assertions. Full-repo PHPStan now reports 0 errors.

File: tests/Unit/inc/ThirdParty/Hostings/StudioPress/clearCacheAfterAccelerator.php (hunk 3 above)
Comment (remyperona) [reply tests/Unit/inc/ThirdParty/Hostings/StudioPress/clearCacheAfterAccelerator.php]: Fixed in 49f248169 — added a PHPStan-resolvable stub `tests/Fixtures/classes/SP_Accel_Nginx_Proxy_Cache_Purge.php` and replaced the `addToAssertionCount()` calls with real assertions. Full-repo PHPStan now reports 0 errors.

File: tests/Unit/inc/ThirdParty/Hostings/StudioPress/cleanAcceleratorCache.php (hunk 4 above)
Comment (remyperona) [reply tests/Unit/inc/ThirdParty/Hostings/StudioPress/cleanAcceleratorCache.php]: Fixed in 49f248169 — added a PHPStan-resolvable stub `tests/Fixtures/classes/SP_Accel_Nginx_Proxy_Cache_Purge.php` and replaced the `addToAssertionCount()` calls with real assertions. Full-repo PHPStan now reports 0 errors.

File: tests/Integration/inc/ThirdParty/Hostings/StudioPress/cleanAcceleratorCache.php (hunk 5 above)
Comment (remyperona) [reply tests/Integration/inc/ThirdParty/Hostings/StudioPress/cleanAcceleratorCache.php]: Fixed in 49f248169 — added a PHPStan-resolvable stub `tests/Fixtures/classes/SP_Accel_Nginx_Proxy_Cache_Purge.php` and replaced the `addToAssertionCount()` calls with real assertions. Full-repo PHPStan now reports 0 errors.

**Comment by remyperona** [comment]
> [!IMPORTANT]
> **On hold — do not merge yet.** Pending confirmation of whether StudioPress Accelerator is still a supported integration.

The migration itself is complete, green (all CI passing), QA-passed, and lead-review-approved — it strictly preserves current behavior. However, before merging we want to resolve an open question raised during review:

**Is StudioPress Accelerator still in use, or fully deprecated since the WP Engine acquisition?**

Signals that it may be dead:
- The integration logic has had **no purposeful, product-driven change since ~2017–2018** — every commit since is an incidental repo-wide sweep (coding standards, capability/sanitization pass, a WPML fix), not StudioPress-Accelerator work.
- No public documentation could be found for the `SP_Accel_Nginx_Proxy_Cache_Purge` class, the `sp_accel_nginx_proxy_cache_purge` global, or the `sp-accel-purge-url` / `sp-accel-purge-theme` nonce actions.

**What unblocks this PR:**
- If telemetry shows ~0 installs with `SP_Accel_Nginx_Proxy_Cache_Purge` present (or WP Engine confirms Accelerator is retired) → the integration is dead and the right move is to **delete it entirely** (drop the `require` + file, no new subscriber), not migrate it. This PR would be repurposed or closed in favor of a removal.
- If it's still supported → **merge this PR as-is** (safe, behavior-preserving, and it unblocks the resolver optimization in #6418).

Marking as draft until that decision is made.
